chutney/server/src/main/java/com/chutneytesting/execution/infra/storage/DatabaseExecutionJpaRepository.java

```diff
@@ -31,7 +31,21 @@ public interface DatabaseExecutionJpaRepository extends JpaRepository<ScenarioEx
 
     List<ScenarioExecutionEntity> findByScenarioIdOrderByIdDesc(String scenarioId);
 
-    @Query("select max(se.id), se.scenarioId from SCENARIO_EXECUTIONS se where se.scenarioId in :scenarioIds AND se.status != 'NOT_EXECUTED' group by se.scenarioId")
+    @Query("""
```

Comment:
I think i would add another method here to fetch the last executions with a given status (or a list of status) and use it in repository along this one to merge by hand the last executions' ids with the last running executions' ids.

Reply:
The main purpose of this method isn't to return the last executions with a given status.
It returns the last running exec **if available**, otherwise the last exec